mne/io/tests/test_constants.py

```diff
@@ -0,0 +1,167 @@
+                    assert fiff_version is None
+                    fiff_version = line.split()[-1]
+                if (line.startswith('#') or line.startswith('alias') or
+                        len(line) == 0):
+                    continue
+                line = line.split('"')
+                assert len(line) in (1, 2, 3)
+                desc = '' if len(line) == 1 else line[1]
+                line = line[0].split()
+                assert len(line) in (2, 3)
+                if len(line) == 2:
+                    kind, id_ = line
+                else:
+                    kind, id_, tagged = line
+                    assert tagged in ('tagged',)
+                id_ = int(id_)
+                if id_ not in iod_dups:
+                    assert id_ not in iod
+                iod[id_] = [kind, desc]
+    # Tags (MEGIN)
+    tags = dict()
+    with open(op.join(tmpdir, 'DictionaryTags.txt'), 'rb') as fid:
+        for line in fid:
+            line = line.decode('ISO-8859-1').strip()
+            if (line.startswith('#') or line.startswith('alias') or
+                    line.startswith(':') or len(line) == 0):
+                continue
+            line = line.split('"')
+            assert len(line) in (1, 2, 3), line
+            desc = '' if len(line) == 1 else line[1]
+            line = line[0].split()
+            if int(line[1]) in tag_bads:
+                assert len(line) == 2
+                kind, id_ = line
+                dtype, unit = '-', '-'
+            else:
+                assert len(line) == 4, line
+                kind, id_, dtype, unit = line
+            id_ = int(id_)
+            val = [kind, dtype, unit]
+            assert id_ not in tags, (tags.get(id_), val)
+            tags[id_] = val
+    # Tags (MNE)
+    with open(op.join(tmpdir, 'DictionaryTags_MNE.txt'), 'rb') as fid:
+        for li, line in enumerate(fid):
+            line = line.decode('ISO-8859-1').strip()
+            # ignore continuation lines (*)
+            if (line.startswith('#') or line.startswith('alias') or
+                    line.startswith(':') or line.startswith('*') or
+                    len(line) == 0):
+                continue
+            # weird syntax around line 80:
+            if line in ('/*', '"'):
+                continue
+            line = line.split('"')
+            assert len(line) in (1, 2, 3), line
+            if len(line) == 3 and len(line[2]) > 0:
+                l2 = line[2].strip()
+                assert l2.startswith('/*') and l2.endswith('*/'), l2
+            desc = '' if len(line) == 1 else line[1]
+            line = line[0].split()
+            assert len(line) == 3, (li + 1, line)
+            kind, id_, dtype = line
+            unit = '-'
+            id_ = int(id_)
+            val = [kind, dtype, unit]
+            if id_ not in tag_dups:
+                assert id_ not in tags, (tags.get(id_), val)
+            tags[id_] = val
```

Comment:
you cannot fuse / merge the 3 with code blocks to avoid the duplication?

Reply:
The formats and exceptions differ so it would lead to more branching. They have more things different than they have in common so I don't think it would make things clearer.